common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java

```diff
@@ -0,0 +1,279 @@
+import java.util.StringTokenizer;
+import java.util.UUID;
+
+import static com.microsoft.identity.common.adal.internal.util.StringExtensions.isNullOrBlank;
+import static com.microsoft.identity.common.adal.internal.util.StringExtensions.urlFormDecode;
+
+public abstract class MicrosoftAuthorizationRequest extends AuthorizationRequest {
+    /**
+     * Serial version id.
+     */
+    private static final long serialVersionUID = 6873634931996113294L;
+
+    /* Constants */
+    private static final String TAG = MicrosoftAuthorizationRequest.class.getSimpleName();
+    public static final String ENCODING_UTF8 = "UTF_8";
+    public static final String CODE_CHALLENGE = "code_challenge";
+    public static final String CODE_CHALLENGE_METHOD = "code_challenge_method";
+    public static final String QUERY_PROMPT = "prompt";
+    public static final String QUERY_PROMPT_VALUE = "login";
+    public static final String LOGIN_HINT = "login_hint";
+
+    public static final String LIB_ID_PLATFORM = "x-client-SKU";
+    public static final String LIB_ID_VERSION = "x-client-Ver";
+    public static final String LIB_ID_CPU = "x-client-CPU";
+    public static final String LIB_ID_OS_VER = "x-client-OS";
+    public static final String LIB_ID_DM = "x-client-DM";
+
+    /**
+     * Required.
+     */
+    private URL mAuthority;
+    /**
+     * Required value.
+     * <p>
+     * Passed in from ADAL/MSAL after authority verification.
+     */
+    private String mAuthorizationEndpoint;
+    /**
+     * Can be used to pre-fill the username/email address field of the sign-in page for the user, if you know their username ahead of time.
+     */
+    private String mLoginHint;
+    /**
+     * Correlation ID.
+     */
+    private UUID mCorrelationId;
+    /**
+     * Used to secure authorization code grants via Proof Key for Code Exchange (PKCE) from a native client.
+     */
+    private PkceChallenge mPkceChallenge;
+    /**
+     * Extra query parameters.
+     */
+    private String mExtraQueryParam;
+    /**
+     * The version of the calling library.
+     */
+    private String mLibraryVersion;
+
+    /**
+     * Constructor of MicrosoftAuthorizationRequest.
+     */
+    public MicrosoftAuthorizationRequest(final String responseType,
+                                         @NonNull final String clientId,
+                                         @NonNull final String redirectUri,
+                                         final String state,
+                                         final Set<String> scope,
+                                         @NonNull final URL authority,
+                                         @NonNull final String authorizationEndpoint,
+                                         final String loginHint,
+                                         final UUID correlationId,
+                                         final PkceChallenge pkceChallenge,
+                                         final String extraQueryParam,
```

Comment:
No javadoc expressing the expectations around format for the extra qp.... I would more intuitively expect this parameter to be a `Map` than a `String`....

are they encoded?
space delimited?
more than one are acceptable?